src/System.Memory/tests/ReadOnlySpan/CopyTo.cs

```diff
@@ -188,5 +189,32 @@ public static void CopyToLargeSizeTest(long bufferSize)
                 }
             }
         }
+
+        [Fact]
+        public static void CopyToVaryingSizes()
+        {
+            var rng = new Random();
+            byte[] inputBuffer = new byte[2048];
+            byte[] outputBuffer = new byte[2048];
+
+            // Test all inputs from size 0 .. 2048 (inclusive) to make sure we don't have
+            // gaps in our Memmove logic.
+            for (int i = 0; i <= inputBuffer.Length; i++)
+            {
+                // Arrange
+
+                rng.NextBytes(inputBuffer);
+                Array.Clear(outputBuffer, 0, outputBuffer.Length);
+
+                // Act
+
+                new ReadOnlySpan<byte>(inputBuffer, 0, i).CopyTo(outputBuffer);
+
+                // Assert
+
+                Assert.Equal(inputBuffer.Take(i), outputBuffer.Take(i)); // src successfully copied to dst
```

Comment:
nit: Can we use SequenceEqual/Slice to check for equality here instead of Linq?

Same with the check that no other part of the destination was overwritten.

```C#
Assert.True(new Span<byte>(inputBuffer).Slice(0, i).SequenceEqual(new Span<byte>(outputBuffer).Slice(0, i)));
```

Not sure how much using Linq impacts test time, especially given we loop from 0 to 2048.